src/uucore/src/lib/mods/line_ending.rs

```diff
@@ -0,0 +1,42 @@
+/// Aims to provide consistent newline/zero terminator handling
+use std::fmt::Display;
+
+#[repr(u8)]
+#[derive(Clone, Copy, Debug, Default, PartialEq)]
+/// Line terminator used for printing and parsing
+pub enum LineEnding {
+    #[default] Newline = b'\n',
+    Nul = 0,
+    Space = b' ',
+    None = 8, // abuse backspace \b to encode None
+}
+
+impl Display for LineEnding {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            Self::Newline => writeln!(f),
+            Self::Nul => write!(f, "\0"),
+            Self::Space => write!(f, " "),
+            Self::None => std::fmt::Result::Ok(()),
+        }
+    }
+}
+
+impl From<LineEnding> for u8 {
+    fn from(line_ending: LineEnding) -> Self {
+        match line_ending {
+            LineEnding::None => panic!("Cannot convert LineEnding::None to u8"),
+            _ => line_ending as Self,
+        }
+    }
+}
+
+impl From<bool> for LineEnding {
+    fn from(is_zero_terminated: bool) -> Self {
```

Comment:
I don't like that this is `from`. A more descriptive name would be nice.

Reply:
I haven't invented this conversion, but simply extracted it from `src/uu/comm/src/comm.rs` 🙂

Reply:
The naming for functions needs to be a bit clearer in `uucore` than in a single util, I suppose. We can keep the method but just rename it. Maybe `from_zero_flag`?

Reply:
Makes sense, yeah. Done.